cmd/day1-utility/main.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/rh-ecosystem-edge/kernel-module-management/pkg/mcproducer"
+)
+
+var (
+	GitCommit = "undefined"
+	Version   = "undefined"
+)
+
+func main() {
+	var (
+		mcName       = flag.String("machineConfigName", "", "name of the MachineConfig to be created")
+		mcpPool      = flag.String("machineConfigPoolName", "", "MachineConfigPool that should be referenced by MachineConfig")
+		moduleImage  = flag.String("kernelModuleImage", "", "kernel module container image")
+		kernelModule = flag.String("kernelModule", "", "kernel module that should be loaded by MachineConfig")
+	)
+
+	flag.Parse()
+
+	if flag.NFlag() != 4 {
+		fmt.Println("Not enough parameters are provided", "num parameters", flag.NFlag(), "git commit", GitCommit, "version", Version)
+		flag.Usage()
+		os.Exit(1)
+	}
```

Comment:
If those values are mandatory, then they should be arguments, not flags.

Reply:
flags package does not support mandatory parameters. By checking the number of flag we are actually making them mandatory